src/k8s/pkg/k8sd/api/certificates_refresh.go

```diff
@@ -130,20 +133,54 @@ func refreshCertsRunControlPlane(s state.State, r *http.Request, snap snap.Snap)
 		return response.InternalError(fmt.Errorf("failed to generate control plane kubeconfigs: %w", err))
 	}
 
-	if err := snaputil.RestartControlPlaneServices(r.Context(), snap); err != nil {
-		return response.InternalError(fmt.Errorf("failed to restart control plane services: %w", err))
-	}
+	// NOTE: Restart the control plane services in a separate goroutine to avoid
+	// restarting the API server, which would break the k8sd proxy connection
+	// and cause missed responses in the proxy side.
+	readyCh := make(chan error)
+	go func() {
+		// NOTE: Create a new context independent of the request context to ensure
+		// the restart process is not cancelled by the client.
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
Can this be too short for slow/busy machines? Restarting especially the kube-apiserver may take some time. 
(I don't think this is an issue - just thinking out loud.)

Reply:
Yeah, you're right. Specially in hyperconverged and busy deployments this could take a long time. I'm going to bump the timeout to 5 minutes, that should be enough.